tensorrt_llm/_torch/cute_dsl_kernels/blackwell/conv/dense_implicit_gemm_fprop.py

```
@@ -0,0 +1,2034 @@
+    LowerPadD, LowerPadH, LowerPadW = lower_padding_dhw
+    DilD, DilH, DilW = dilation_dhw
+    Z = ((D + UpperPadD + LowerPadD - DilD * (T - 1) - 1) // Sd) + 1
+    P = ((H + UpperPadH + LowerPadH - DilH * (R - 1) - 1) // Sh) + 1
+    Q = ((W + UpperPadW + LowerPadW - DilW * (S - 1) - 1) // Sw) + 1
+    return Z, P, Q
+
+
+def create_cute_tensor(
+    source_f32_tensor: torch.Tensor,
+    dtype: Type[cutlass.Numeric],
+    leading_dim: int = None,
```

Ruff (0.16.1)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)